SCRUM-255 design: add music component

gdaegeun539 opened the pull request:
### Please check if the PR fulfills these requirements
- [x] The commit message follows our guidelines
<!-- - [ ] Tests for the changes have been added (for bug fixes / features) ~~  -->
<!-- - [ ] Docs have been added / updated (for bug fixes / features) -->


## What kind of change does this PR introduce?
> Select your one and delete others

- Implement design


# What is the current behavior?
>  You can also link to an open issue here

SCRUM-255


# What is the new behavior (if this is a feature change)?

Music 컴포넌트를 구현합니다.
해당 컴포넌트가 사용되는 [5가지 형태](https://www.figma.com/design/SzlnIcgTVChfLWd5dNOOox?node-id=2500-29536#1542919162)를 모두 포함합니다.


## Does this PR introduce a breaking change? (What changes might users need to make in their application due to this PR?)

No breaking changes.
기존 컴포넌트에 신규 컴포넌트를 사용하는 방식으로 사용하지만, 이번에 제작한 신규 컴포넌트의 사용법과 동일합니다.


## ScreenShots (If needed)

<p>
  <img src="https://github.com/user-attachments/assets/43c2dfe5-3eb1-4e0d-96c4-2685e089f9f3" width="300" />
  <img src="https://github.com/user-attachments/assets/306c258e-11c2-4ff4-91b9-4b9a169d575d" width="300" />
</p>
<p>
  NoteComponent에 이 컴포넌트를 사용해도 형태가 변형되지 않습니다. <br/>
  <img src="https://github.com/user-attachments/assets/31215d3d-0656-4970-92be-b3ac47d53a1e" width="300" />
</p>


## Other information:

마이페이지 요소를 단순히 표시하는 부분과는 약간 떨어진 컴포넌트인데, 구현한 부분은 넣는 게 맞는 듯 해 이 PR까지는 제출합니다.

gdaegeun539 committed:
refactor: basic music component extract

- 기존 NoteComponent.kt에서 음악 부분만 컴포넌트 파일로 분리합니다.
- 작업이 더 필요한 상태입니다.

gdaegeun539 committed:
chore: modify ai agent guidelines

gdaegeun539 committed:
fix/design: change wrong font size of caption1

gdaegeun539 committed:
fix/design: change wrong color of light brand secondary

gdaegeun539 committed:
design: add song_list, plus icon asset

gdaegeun539 committed:
design: add empty album image asset

gdaegeun539 committed:
design: add music component's all design with data

- Music 컴포넌트가 나타내야 하는 5가지의 상태를 sealed class로 추가
- MusicComponent의 Slot API 기반 레이아웃을 제작
  - 각 컴포넌트 별로 맞는 레이아웃 컨텐츠를 추가

gdaegeun539 committed:
refactor: update code lint

hyunjung-choi reviewed (commented):
다크모드는 구현 생략하신건가요 !?

gdaegeun539 commented:
> 다크모드는 구현 생략하신건가요 !?

다크모드 프리뷰 추가해놓겠습니다. 내부에는 `LocalFeelinColors.current`로 접근해서 사용하고 있어서 대응되어 있어요.

gdaegeun539 committed:
refactor: add dark mode preview at music component

gdaegeun539 commented:
다크모드 프리뷰 추가했습니다. @hyunjung-choi 현정님 코테랑 면접 화이팅입니당

hyunjung-choi commented on app/src/main/java/com/lyrics/feelin/presentation/view/component/music/MusicComponentData.kt:
```diff
@@ -0,0 +1,63 @@
+    val imageUrl: String
+    val songName: String
+    val artistName: String
+}
```
얘는 파일을 분리하는게 나을까요..? 🤔

gdaegeun539 replied:
음 이 인터페이스의 역할은 아래 `MusicComponentState`에 공통된 세부 데이터가 들어갈 때 제약을 하는 역할 밖에 없어서 굳이 분리하기는 그래요.

이 인터페이스는 private으로 두는 건 어떨까요?
(이렇게 하면 뷰 데이터용 sealed 인터페이스의 이름을 detekt 규칙에 따라 `MusicComponentData`로 바꿀 거긴 해요.)

hyunjung-choi replied:
오 좋아요 !

hyunjung-choi commented on app/src/main/java/com/lyrics/feelin/core/designsystem/icon/Icons.kt:
```diff
@@ -78,3 +78,15 @@ val XCircleIcon: Painter
+
+val PlusIcon: Painter
+    @Composable
+    get() = painterResource(id = R.drawable.plus)
```
벡터 아이콘(`.xml`)은 `ImageVector`로 통일하구
Bitmap 또는 PNG 리소스만 `Painter`로 사용하기로 해요 ~ !!

---

사실 . . .
**`Icon` 내부에서 `ImageVector`가 `Painter`로 변환**되긴 합니다 ! 😅
하지만 **`ImageVector`는 그림 정보(Color, Path)만 저장**되어 있기도 하고, 
해당 리소스가 벡터인지 아닌지 코드만 봐도 알 수 있도록 명확하게 표현하고자 해서 제안드립니다 ! ㅎㅎ

gdaegeun539 replied:
> 하지만 ImageVector는 그림 정보(Color, Path)만 저장되어 있기도 하고,
> 해당 리소스가 벡터인지 아닌지 코드만 봐도 알 수 있도록 명확하게 표현하고자 해서

오호 하나 알아갑니다. 👍 이건 이 의견대로 가는 게 좋을 것 같아요.
노션에도 이런 논의 내용을 간략하게 페이지로 기록해 두면 나중에 보기 좋을 것 같습니다!

hyunjung-choi reviewed (commented):
코드가 너무 깔끔해요 . . 👍

gdaegeun539 commented:
> 코드가 너무 깔끔해요 . . 👍

잼민이였는지 클로드였는지는 기억이 안 나지만 공통된 구조를 깔끔하게 가져갈 수 있도록 같이 고민해봤습니당
그렇게 느끼셨다니 좋네요...

gdaegeun539 committed:
chore: modifiy ai agent rules

gdaegeun539 committed:
refactor: Use ImageVector for all vector icons

Replace Painter implementation with ImageVector for remaining vector icons in Icons.kt to ensure consistency and clarity.

- Convert CaretIcon, EmptyImageDarkIcon, etc. to ImageVector

- Remove unused Painter imports

gdaegeun539 committed:
refactor: update code lint

gdaegeun539 committed:
chore: add commit workflow for antigravity agent

gdaegeun539 committed:
refactor: Migrate vector icons to ImageVector type

Update Icons.kt to return ImageVector instead of Painter for vector assets.

Resolve type mismatch compilation errors in LoginScreen, MyPageScreen,

and UI components by replacing painter parameter with imageVector.

hyunjung-choi reviewed (approved):
`interface`부분 수정 부탁드립니다 ! 고생하셨습니다 !! 🙇‍♀️

gdaegeun539 committed:
refactor: Rename MusicComponentState to MusicComponentData

Rename the sealed interface from MusicComponentState to MusicComponentData
for better naming consistency with component data patterns.

- Rename MusicComponentState to MusicComponentData in MusicComponentData.kt
- Update all references in MusicComponent.kt
- Update import and usage in NoteComponent.kt
- Make MusicData interface private as it is only used internally

gdaegeun539 commented:
> `interface`부분 수정 부탁드립니다 ! 고생하셨습니다 !! 🙇‍♀️

감사합니다! 수정 완료했고 머지하겠습니다.
